client/src/lib/graph/createRelatedGraph.ts

```diff
@@ -0,0 +1,52 @@
+import { Node, Edge } from "../api/api_pb";
+
+const createRelatedGraph = (
+  nodes: Node[],
+  edges: Edge[],
+  selectedWordId: string | null,
+  depth: number
+): {
+  nodes: Node[],
+  edges: Edge[],
+} => {
+  // 選択されたノードが不正な場合には空のグラフを返す
+  if (selectedWordId === null) return {nodes: [], edges: []}
+  
+  const resultNodeIds = new Set<string>()
+  const resultEdges = new Set<Edge>()
+  
+  const nodeRelation = nodes.reduce<Record<string, [Edge, string][]>>((acc, node) => {
+    acc[node.id] = []
+    return acc
+  }, {})
```

Comment:
こういう書き方もできますがどちらでも良いかな
```suggestion
  const nodeRelation = Object.fromEntries(nodes.map(node => [node.id, []]))
```